### PR TITLE
Device list (nmstatectl show)

### PR DESCRIPTION
**Pull request by EdDev**
List the network devices with basic properties.

### CODE REVIEW

**Comment by tyll** [comment]
Could you maybe sqash the tox.ini changes and move the commits together with the travis change to the beginning of the change?

**Comment by EdDev** [comment]
The last two commits were just an attempt to fix the tests, but it turns out it is a futile battle.
The tests that came in this PR are integration/functional ones, requiring NM service and other helper packages. We will not be able to have them on the travis-ci for the time being.
I therefore need to convert all tests to unit ones, mocking out the gi package suite.

File: libnmstate/netinfo.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#
+from __future__ import absolute_import
+
+from libnmstate import nmclient
+from libnmstate import validator
+
+
+def show():
+    report = interfaces()
+    validator.verify(report)
```
[the remark this answers is not in the document]
Comment (EdDev) [reply libnmstate/netinfo.py]: Why is that? The validation should be performed on both input and output to assure it confirms with the expected schema.

File: tests/ctl/compat.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#
```
[the remark this answers is not in the document]
Comment (EdDev) [reply tests/ctl/compat.py]: We should, but I had no good solution at the moment. We probably need a testlib package which can be used by all tests.
We will have to handle this in near future.

**Pull request merged**